docs: moving best examples into subdirs

staceysv opened the pull request:
* added app, reference, tutorial under examples
* moves some of the best examples into these

scottire commented:
- Can we move `app` to `experimental`, I'll let Tristan know
- Can we move `control` `layout` `panel/plot` and `viz` under `reference`.
- Can we rename `tutorial` to `tutorials`
- `apps` is very empty now and we don't have immediate plans to improve that
-- Is there anything that can go in there now from `experimental`/`app`?
-- If not, can we move the quickstart in `apps` to `tutorial` and delete `apps` for now.
--- In the future, this will be where we can keep instructions to use different templates

staceysv commented:
- yes, let's put app/ under experimental & all the others under reference—thanks for doing that! eventually may want to re-org and flatten.
- renamed tutorial => tutorials, removed all the other empty folders
- love what you did with getting started, numbers, and apps! i personally don't like the "app" / "apps" name because it doesn't mean anything to me—let's go with this structure and see what else we can improve/add.

scottire commented:
> i personally don't like the "app" / "apps" name because it doesn't mean anything to me—let's go with this structure and see what else we can improve/add
agreed, I was torn between that and `gallery`. Went with gallery instead

staceysv commented on integration_test/cypress/e2e/testlib.ts:
@dannygoldstein could you double-check this? minor change but could affect tests

staceysv commented on weave-js/src/components/Panel2/PanelBoolean/Component.tsx:
@dannygoldstein also a quick check here—just adding an id to this component so we can use a PanelBoolean in notebooks and pass notebook tests

staceysv commented:
I have read the CLA Document and I hereby sign the CLA
